src/sprightly.ts

```diff
@@ -1,54 +1,79 @@
 import { readFileSync } from "fs"
 import path from "path"
-import KeyNotFoundError from "./errors/KeyNotFoundError"
-import PathNotFoundError from "./errors/PathNotFoundError"
 
 interface Data {
   [key: string]: string | number | Array<string | number | Data> | Data
 }
 
+class KeyNotFoundError extends Error {
+  readonly key: string
+  readonly src: string
+  constructor(key: string, src: string) {
+    const message = `Key "${key}" not found in "data" at ${src}`
+    super(message)
+    this.key = key
+    this.src = src
+  }
+}
+
+class PathNotFoundError extends Error {
+  readonly component: string
+  readonly src: string
+  constructor(component: string, src: string) {
+    const message = `Component "${component}" was not found in ${src}`
+    super(message)
+    this.component = component
+    this.src = src
+  }
+}
+
 interface Options {
   keyFallback?: string
   throwOnKeyNotfound?: boolean
   cache?: boolean
 }
 
-const cache = new Map<string, string>()
-
-const defaultOptions = {
-  keyFallback: "",
-  throwOnKeyNotfound: false,
-  cache: false,
-}
-
 function parse(file: string, data: Data, options: Options) {
```

Comment:
You need to add the `filename` parameter to this that is mentioned here https://github.com/obadakhalili/sprightly/pull/4#discussion_r978353593.